### ISSUE TITLE
The Text Extractor can't handle empty strings

### ISSUE THREAD

**Issue by Michael-F-Bryan** [hotg-ai/proc-blocks#9]
I was playing around with the web runtime when I noticed passing an empty string to the Bert rune will trigger [this assertion](https://github.com/hotg-ai/proc-blocks/blob/2ed797ef5d5a9703074f244dc60ac262c052120d/text_extractor/src/lib.rs#L43-L48).

In this case, it looks like `end_index` and `start_index` are both zero. 

I'm guessing we'll also see this happen when the model is unable to figure out the answer to your question.

**Comment by Mohit0928** [comment]
Thanks, @Michael-F-Bryan, for finding it out. I will correct it.

**Comment by Michael-F-Bryan** [comment]
If it helps, here is what we log when the panic occurs:

```json
{
    "level": "ERROR",
    "message": "panicked at 'Start index: 0 is greater than or equal to end index: 0', /Users/mohit/.cargo/git/checkouts/proc-blocks-49e4d24512f5d324/1e53d6f/text_extractor/src/lib.rs:44:13",
    "target": "hotg_runicos_base_wasm",
    "module_path": "hotg_runicos_base_wasm",
    "file": "/Users/mohit/.cargo/registry/src/github.com-1ecc6299db9ec823/hotg-runicos-base-wasm-0.10.0/src/lib.rs",
    "line": 52
}
```

**Comment by Michael-F-Bryan** [comment]
I believe @Mohit0928 created #10 to resolve this, but we're running into the panics [in the `text_extractor` proc block](https://github.com/hotg-ai/proc-blocks/blob/7c78b515048dc660b2d516c21a9b49c01d804524/text_extractor/src/lib.rs#L41-L48), while that PR's changes were added to the `tokenizer` proc block.

I think we need to rewrite [these lines](https://github.com/hotg-ai/proc-blocks/blob/7c78b515048dc660b2d516c21a9b49c01d804524/text_extractor/src/lib.rs#L41-L48) in a way that lets us handle zero, one, or multiple sentences as an output. Possibly using `zip()` to iterate over pairs of start/end logits, constructing sentences until we get an invalid pair (e.g. `start_index = end_index = 0` because there is no answer).

**Comment by Michael-F-Bryan** [comment]
No, that's not what we want to do.

The underlying problem is that the model was unable to generate an answer for the provided inputs so the text extractor crashed the Rune (the logits tensor passed to the text extractor was all zeroes). Your "sentence 1 is empty" assertion is in the tokenizer so it won't help prevent this situation.

Another thing to keep in mind is the text passed to a tokenizer is controlled by the user. in general, you should try to handle bad user input graciously (e.g. by setting the tokenizer output to all zeroes) so unexpected input won't crash the runtime. There's a pretty common saying for this:

> *Be liberal in what you accept, and conservative in what you send* - [the Robustness Principle](https://en.m.wikipedia.org/wiki/Robustness_principle)

Also, assertions are mainly be used for unrecoverable errors that were most probably caused by a programming bug (e.g. non-finite floats due to a divide by zero). For errors you can expect to see frequently like empty input, we should try to return a sensible default value like an empty string.
